src/main/webapp/cdn/blockly/generators/propc/i2c.js

```diff
@@ -102,7 +102,7 @@ Blockly.propc.i2c_in = function () {
     var device = Blockly.propc.valueToCode(this, 'DEVICE', Blockly.propc.ORDER_ATOMIC);
     var size = Blockly.propc.valueToCode(this, 'SIZE', Blockly.propc.ORDER_ATOMIC);
     var address = Blockly.propc.valueToCode(this, 'REGISTER', Blockly.propc.ORDER_ATOMIC);
-    var data = Blockly.propc.valueToCode(this, 'DATA', Blockly.propc.ORDER_NONE);
+    var data = Blockly.propc.valueToCode(this, 'DATA', Blockly.propc.ORDER_NONE) || '';
```

Comment:
I'd the user doesn't add a block, it provides a null that can be tested
for.  If it's not there, and the user doesn't add a block, it throws an
"undefined"

On Sep 3, 2016 12:09 PM, "Parallax Git Administrator" <
notifications@github.com> wrote:

> In src/main/webapp/cdn/blockly/generators/propc/i2c.js
> <https://github.com/parallaxinc/BlocklyProp/pull/691#discussion_r77441606>
> :
>
> > @@ -102,7 +102,7 @@ Blockly.propc.i2c_in = function () {
> >      var device = Blockly.propc.valueToCode(this, 'DEVICE', Blockly.propc.ORDER_ATOMIC);
> >      var size = Blockly.propc.valueToCode(this, 'SIZE', Blockly.propc.ORDER_ATOMIC);
> >      var address = Blockly.propc.valueToCode(this, 'REGISTER', Blockly.propc.ORDER_ATOMIC);
> > -    var data = Blockly.propc.valueToCode(this, 'DATA', Blockly.propc.ORDER_NONE);
> > +    var data = Blockly.propc.valueToCode(this, 'DATA', Blockly.propc.ORDER_NONE) || '';
>
> What does || '' do?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/parallaxinc/BlocklyProp/pull/691/files/2539e4abfda567b0b0f07ce0b6d7bcf6c56e9c9f#r77441606>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AS0quM3PU_zZXVJ1v4SATSA6XU7QOe7Tks5qmcX5gaJpZM4J0Xpr>
> .
>